### ISSUE TITLE
NullPointerException in MultiOccurrenceToMultiOccurrence.mapFields

### ISSUE THREAD

**Issue by GoogleCodeExporter** [vadbch/orika#85]
```
What steps will reproduce the problem?
1. Run the attached program.


What version of the product are you using? On what operating system?
1.4.1, java 1.6, windows 7

    public static void main(String[] args)
    {
        MapperFactory f = new DefaultMapperFactory.Builder().build();
        f.classMap(MapContainer.class, Map.class)
            .field("map{value}", "{key}")
            .register();
        final MapperFacade facade = f.getMapperFacade();
        final Map<Object, Object> dest = new HashMap<Object, Object>();
        final Map<Object, Object> src = new HashMap<Object, Object>();
        src.put("xyz", "123456");
        facade.map(new MapContainer(src), dest);
        System.out.println(dest);
    }

where MapContainer is defined like this:

public final class MapContainer
{
    public final Map<?, ?> map;

    public MapContainer(final Map<?, ?> map)
    {
        this.map = map;
    }
}
```

Original issue reported on code.google.com by `cona...@gmail.com` on 28 Feb 2013 at 5:12

Attachments:
- [Main.java](https://storage.googleapis.com/google-code-attachments/orika/issue-85/comment-0/Main.java)
- [MapContainer.java](https://storage.googleapis.com/google-code-attachments/orika/issue-85/comment-0/MapContainer.java)

**Comment by GoogleCodeExporter** [comment]
```

Please look at this example if it fit your need,

My first remarks about your test, there is no type information on 
MapConainer.map


```

Original comment by `elaat...@gmail.com` on 1 Mar 2013 at 10:40

Attachments:
- [Main.java](https://storage.googleapis.com/google-code-attachments/orika/issue-85/comment-1/Main.java)

**Comment by GoogleCodeExporter** [comment]
```
The problem isn't so much that the code doesn't work (MapContainer turns out to 
be completely unnecessary with inline properties anyhow), but that the error I 
get isn't particularly helpful. Adding some mapping validation code might be 
helpful.

I did also try with a generic MapContainer<K,V> (that's the code that's 
actually attached), which also seemed not to work.

```

Original comment by `cona...@gmail.com` on 6 Mar 2013 at 6:22

**Comment by GoogleCodeExporter** [comment]
```
Ok so the first point is about error reporting ?

Generics type information are not available in Runtime due to Java 
type-erasure. This is why we introduce Orika custom Type which is a variant of 
super type token solution 

http://gafter.blogspot.co.uk/2006/12/super-type-tokens.html


If you want to use directly Map you can try with MapperFacade.mapAsMap 





```

Original comment by `elaat...@gmail.com` on 7 Mar 2013 at 2:11

**Comment by GoogleCodeExporter** [comment]
```
Makes sense. I do think the error reporting could be more helpful, which is 
really what the issue is about, yes.
```

Original comment by `cona...@gmail.com` on 14 Mar 2013 at 3:48

**Comment by GoogleCodeExporter** [comment]
```
Absolutely and this was one the points we try to improve. 
So thanks for the report.
```

Original comment by `elaat...@gmail.com` on 14 Mar 2013 at 4:07
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
a fix has been uploaded to address this particular issue ( actually there were 
a few uncovered by this one).
Please take a look at the current master...
```

Original comment by `matt.deb...@gmail.com` on 19 Jun 2013 at 6:38
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.4.3 release.
```

Original comment by `matt.deb...@gmail.com` on 7 Jul 2013 at 4:03
- Changed state: **Fixed**
